envs/main.yml

```diff
@@ -17,6 +17,7 @@
   channels:
     - bioconda
     - conda-forge
+    - r
```

Comment:
You do not need to add `r` here - every *snakemake* rule will be executed in its own separate v-env.

As I see - the problem is only with the unit tests for the scripts. How about we add to the CI building all conda v-envs based on the workflow YAML files and then - before each unit test we activate a given, dedicated v-env. Could you please modify the travis CI a little?